### PR TITLE
Docs release: adding portal-wide metadata

### PR DESCRIPTION
**Pull request by jashapiro**
## Purpose

This docs release adds a description of the portal-wide metadata, including updates to the changelog.

Also included are updates to the `alevin-fry` publication, but that did not seem worthy of a changelog entry. Correct me if I am wrong there!

Oh, and changes to spellcheck workflow.

### Issue addressed

#321 

## Checklist for publishing to `main`

- [x] This branch contains all commits/content ready to be released.
- [x] The most recent CHANGELOG entry includes a date.
If this release required a CHANGELOG entry, it should be today's date.

### CODE REVIEW

**Comment by jashapiro** [comment]
Realizing this has to hold off due to https://github.com/AlexsLemonade/scpca-portal/issues/898

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻 

@davidsmejia, please merge this PR when portal-wide metadata is up on prod.

**Pull request merged**
